pkg/controller/aws/aws_helper.go

```diff
@@ -94,18 +88,37 @@ func UpdateNetworkLoadBalancer(loadBalancerDNS string, serviceNameTagValue strin
 	return true, nil
 }
 
-// newAPIClient obtains an AWS session and initiates the needed AWS clients.
-func newAPIClient(id string, secret string, region string) (*APIClient, error) {
+// newAWSConfig generates an AWS config.
+func newAWSConfig() *aws.Config {
+
+	awsRegion := awsDefaultRegion
 
-	// Get AWS config
-	awsConfig := &aws.Config{
-		Region:      aws.String(region),
-		Credentials: credentials.NewStaticCredentials(id, secret, ""),
+	if os.Getenv("AWS_REGION") == "" {
+		log.Info("Empty AWS_REGION, using default value", "awsRegion", awsDefaultRegion)
+	}
```

Comment:
`awsRegion` is never set to the env variable `AWS_REGION` so I guess the code would only work for the default region

Reply:
Good catch, fixed in https://github.com/3scale/aws-nlb-helper-operator/commit/af64c3c860d3655005a25e08767a03da9b5c3328